src/primitives/block.cpp

```diff
@@ -10,21 +10,74 @@
 #include <utilstrencodings.h>
 #include <crypto/common.h>
 #include <streams.h>
+#include <crypto/ethash/helpers.hpp>
+#include <crypto/ethash/include/ethash/progpow.hpp>
+#include <pow.h>
+#include <crypto/randomx/randomx.h>
+
+uint32_t nPowTimeStampActive = 0;
 
 uint256 CBlockHeader::GetHash() const
 {
     return SerializeHash(*this);
 }
 
 #define TIME_MASK 0xffffff80
-uint256 CBlockHeader::GetPoWHash() const
+uint256 CBlockHeader::GetX16RTPoWHash() const
 {
     //Only change every 128 seconds
     int32_t nTimeX16r = nTime&TIME_MASK;
     uint256 hashTime = Hash(BEGIN(nTimeX16r), END(nTimeX16r));
     return HashX16R(BEGIN(nVersion), END(nNonce), hashTime);
 }
 
+uint256 CBlockHeader::GetSha256DPoWHash() const
+{
+    return Hash(BEGIN(nVersion), END(nNonce));
+}
+
+uint256 CBlockHeader::GetProgPowHash() const
```

Comment:
This code should be in `hash.h / hash.cpp` to preserve the primitive as just a data structure. Make similar to `SerializeHash`.

Reply:
Okay will move